### DISCUSSION TITLE
Generate equation number as a span with unique ID

### DISCUSSION THREAD

**Discussion by piszczu4** [KaTeX/KaTeX#3921] [Q&A] [answered]
I'm building a math website and I'm wondering whether it would be possible to somehow generate the equation in such a way that it is possible to copy its ID to be able to create a `link` button which when clicked, it would scroll to that specific equation. Is that possible?

**Comment by mbourne** [comment, marked as answer]
Hello piszczu4

I see you already discovered this discussion about \eqref, which still doesn't seem resolved: https://github.com/KaTeX/KaTeX/issues/2003

At first, I was going to suggest a post-processing step which added an ID to each ```<span class="katex"> ... </span>``` and you could link to those. But if you have many KaTeX expressions on a page and later need to add or remove some of them in the middle of the page, you'd need to update all the links going to all the expressions after the change. It would be a nightmare to maintain.

So why not do the following:

For each expression that you want to link to, add a ```<span>``` around it and give it an ID, some thing like:

<blockquote>Here is Euler's formula &lt;span id="euler_formula"&gt; \( e^{i\pi}+1 = 0 \) &lt;/span&gt; and now for a display integral: &lt;span id="int_sin">$$ \int \sin (x) dx = -cos(x) + C $$&lt;/span&gt;.</blockquote>

Then you can link to those, something like:

<blockquote>Earlier, we saw &lt;a href="#euler_formula"&gt;Euler's formula&lt;/a&gt;  and a random &lt;a href="#int_sin"&gt;sin integral&lt;/a&gt;.</blockquote>

**Reply by piszczu4** [reply]
Yeah thank you, that's the solution I'm going to use in my RichText editor,i.e. allow user to add a label (id) to the equation and number it according to its position within the doc. Bigger problem will be referencing equations from another page, because while `label` of an equation might not change, the page `url` might change when its generated dynamically from for example book chapter name. I mean if I have a page `myPage/geometry/triangles#theorem1`, then when changing the chapter title to `triangles-and-squares`, the `href` will stop work and lead to non-existing page. I need to think how to make it works even with such a cases.
